agimus_controller/agimus_controller/ocp/ocp_traj_tracking_collision_avoidance.py

```diff
@@ -0,0 +1,265 @@
+        constraints = self._get_collision_constraints(active=True)
+        terminal_DAM = crocoddyl.DifferentialActionModelFreeFwdDynamics(
+            self._state, self._actuation, terminal_cost_model, constraints
+        )
+
+        terminal_model = crocoddyl.IntegratedActionModelEuler(terminal_DAM, 1.0)
+        terminal_model.differential.armature = self._robot_models.armature
+        return terminal_model
+
+    def _get_collision_constraints(self, active=True):
+        """Returns the collision constraints model manager."""
+        constraint_model_manager = crocoddyl.ConstraintModelManager(
+            self._state, self.nq
+        )
+        for col_idx in range(len(self._collision_model.collisionPairs)):
+            distance_coll_residual = ResidualDistanceCollision(
+                self._state, 7, self._collision_model, col_idx
+            )
+            collision_constraint = crocoddyl.ConstraintModelResidual(
+                self._state,
+                distance_coll_residual,
+                np.array([self._params.collision_safety_margin]),
+                np.array([np.inf]),
+            )
+            # Adding the constraint to the constraint manager
+            constraint_model_manager.addConstraint(
+                "col_term_" + str(col_idx), collision_constraint, active=active
+            )
+        return constraint_model_manager
+
+    def get_distance_collision_residuals(self) -> Union[npt.NDArray[np.float64], None]:
+        """Return distance collision residual if Crocoddyl's problem use it."""
+        nb_collision_pairs = len(self._collision_model.collisionPairs)
+        if (
+            nb_collision_pairs != 0
+            and self._solver.problem.runningDatas[0].differential.constraints
+            is not None
+        ):
+            coll_residuals = np.zeros((self._params.horizon_size, nb_collision_pairs))
+            for node_idx in range(self._params.horizon_size - 1):
+                constraints_residual_dict = self._solver.problem.runningDatas[
+                    node_idx
+                ].differential.constraints.constraints.todict()
+                for coll_pair_idx, constraint_key in enumerate(
+                    constraints_residual_dict.keys()
+                ):
+                    coll_residuals[node_idx, coll_pair_idx] = constraints_residual_dict[
+                        constraint_key
+                    ].residual.r[0]
+            constraints_residual_dict = self._solver.problem.terminalData.differential.constraints.constraints.todict()
+            for coll_pair_idx, constraint_key in enumerate(
+                constraints_residual_dict.keys()
+            ):
+                coll_residuals[self._params.horizon_size - 1, coll_pair_idx] = (
+                    constraints_residual_dict[constraint_key].residual.r[0]
+                )
+            return coll_residuals
+        else:
+            return None
+
+    def set_reference_weighted_trajectory(
```

Comment:
same as before, maybe we can create a high-level function that will be common for this class and goal_reaching